plugins/java-decompiler/engine/src/org/jetbrains/java/decompiler/modules/decompiler/ExprProcessor.java

```diff
@@ -540,6 +540,7 @@ else if (cn instanceof LinkConstant) {
             }
 
             InvocationExprent exprinv = new InvocationExprent(instr.opcode, invoke_constant, bootstrap_arguments, stack, bytecode_offsets);
+            markPreviousResultUsedIfBoxingExpression(exprlist, exprinv);
```

Comment:
this looks like an overkill, isn't it better to add something like
`if (instance instanceof InvocationExprent) {
            ((InvocationExprent)instance).markUsingBoxingResult();
          }`
right before `instance.toJava...` inside the InvocationExprent.toJava?